### PR TITLE
ChatPanel: add onLinkClick handler for footer

### PR DESCRIPTION
**Pull request by anguyen-yext2**
J=CLIP-432
TEST=auto,manual

tested manually on test site by passing a print statement to the handler and verified that the link is printed when it's clicked. added auto test and ran `npm run test`.

### CODE REVIEW

File: src/components/ChatPanel.tsx [hunk 1]
```diff
@@ -74,6 +74,8 @@ export interface ChatPanelProps
    * can click on instead of typing their own.
    */
   messageSuggestions?: string[];
+  /** A callback which is called when user clicks a footer link. */
```
Comment (yen-tt) [inline src/components/ChatPanel.tsx:77]: shouldn't be just footer link. It should be any link present in Panel in general, which include the MessageBubble and potentially MessageSuggestions.

**Review by yen-tt — COMMENTED** [review]
make sure to resolve errors from the GH lint action (missing deps in useMemo)

File: src/components/ChatPanel.tsx (hunk 1 above)
Comment (anguyen-yext2) [reply src/components/ChatPanel.tsx]: ah ok! I added onLinkClick to MessageBubble as well, but I don't think we can have clickable link within MessageSuggestions (clicking the suggestion = sending the whole msg).

**Review by yen-tt — COMMENTED** [review]
nice! You should probably bump the package version to deploy this change after it's merged. make sure to run `npm run build` after the change.

**Review by yen-tt — APPROVED** [review]

**Pull request merged**